src/hooks/useBifrostWebSocketEffect.ts

```diff
@@ -0,0 +1,83 @@
+import { useEffect } from 'react';
+
+import { bifrostWSClient } from 'services/webSocketClient';
+
+interface IMdc {
+  requestContext: string;
+  processContextVariant: string;
+  level: string;
+  processContext: string;
+}
+
+interface IValue {
+  id: string;
+  timestamp: string;
+  sequence: string;
+  loggerName: string;
+  message: string;
+  stackTrace: string | null;
+  last: boolean;
+  mdc: IMdc;
+  subscriptionTopic: string;
+}
+
+interface IResult {
+  type: string;
+  value: IValue;
+}
+
+interface IWsResponseData {
+  result: IResult;
+  id: number;
+  jsonrpc: string;
+}
+
+interface IUseBifrostWebSocketEffectOptions {
+  buildId?: string;
+
+  // condition based listening, when true, events will be ignored (for example when Build is finished)
+  preventListening?: boolean;
+
+  // debug identification string
+  debug?: string;
+}
+
+/**
+ * WebSocket message listener hook.
+ *
+ * @param callback - Callback method, WebSocket parsed data attribute will be passed in
+ * @param options - See {@link IUseBifrostWebSocketEffectOptions}
+ */
+export const useBifrostWebSocketEffect = (
+  callback: (wsData: any) => void,
```

Comment:
Argument name could match the one which is used later, so in this case `message: string` or more specific `logLine: string` in this case.